BlakeDonn/rancid-tomatillos
# Movie show page & Test

# Rotten Tomatillos PR template
### What's this PR do?  
- A user can re-route from main dashboard to individual movie pages when they click a `PreviewCard`
- Begin setting up more robust tests for: `MoviePage` and `PreviewCard`
- Refactor API calls into separate file
- Rename/refactor some variables for dev empathy
### Where should the reviewer start?  
Main changes are to `MoviePage` and test files
### How should this be manually tested?  
- Run `npm start` to start the server and open the app on your browser - click preview cards on home page to see an individual movie details.
- Run `npm test` to see current partial tests
### Any background context you want to provide?  
- Router / testing in React are new concepts for both of us so we are piecing together what we can from pre-teaching lessons not yet covered in class, lots of googling and talking with classmates/mentors.
### What are the relevant tickets?  
Closes #15;
Closes #32;
Closes #50;
Closes #57
Closes #58;

**In Progress**
#16, #17, #55, #56
### Screenshots (if appropriate)  
### Questions:
- Not sure how to incorporate router/router prop(history) into testing for `PreviewCard` click functionality having been called and other Components testing
- Not sure if `App` test for log-in is implemented correctly - currently passing - is this a false positive?
- #58: Should this be on all elements? Is there a way to refactor so that this only needs to be done once?
